python/src/mozdata/utils.py

```diff
@@ -69,3 +70,72 @@ def spark_list_tables(spark, table_name=None):
             .collect()
         )
     ]
+
+
+def uri_version(uri):
+    """Try to get the table version from a uri
+
+    :param uri: uri from which to attempt version extraction
+    :return: version if found, or None
+    """
+    if uri is not None:
+        version = uri.split("/").pop()
+        if is_version(version):
+            return version
+
+
+class TableInfo:
+    def __init__(self, table_name, version, owner, uri, spark,
+                 ad_hoc_tables_dir, global_tables_dir):
+        self.spark = spark
+        self.table_name = table_name
+        self.version = version
+        self.owner = owner
+        self.uri = uri
+        self.in_catalog = False
+        self.sql_table_name = table_name
+        if version is not None:
+            self.sql_table_name += "_" + version
+
+        if uri is not None:
+            self.version = uri_version(uri)
```

Comment:
This will be covered with the `write_table` method, right?

Reply:
yes